python/core/additions/qgsfunction.py

```diff
@@ -26,43 +24,52 @@
 from qgis._core import QgsExpressionFunction, QgsExpression, QgsMessageLog, QgsFeatureRequest, Qgis
 
 
-def register_function(function, arg_count, group, usesgeometry=False,
-                      referenced_columns=[QgsFeatureRequest.ALL_ATTRIBUTES], handlesnull=False, **kwargs):
+def register_function(
+    function,
+    group,
+    usesgeometry=False,
+    referenced_columns=[QgsFeatureRequest.ALL_ATTRIBUTES],
+    handlesnull=False,
+    **kwargs,
+):
     """
     Register a Python function to be used as a expression function.
 
-    Functions should take (values, feature, parent) as args:
-
-    Example:
-        def myfunc(values, feature, parent):
-            pass
+    The function signature may contains special parameters:
+    - context: the QgsExpressionContext-related to the current evaluation
+    - feature: the QgsFeature-related to the current evaluation
+    - parent: the QgsExpressionFunction parent
 
-    They can also shortcut naming feature and parent args by using *args
-    if they are not needed in the function.
+    If those parameters are present in the function signature, they will be automatically passed to the function,
+    without the need to specify them in the expression.
 
-    Example:
-        def myfunc(values, *args):
-            pass
+    If the only other parameter in the signature is called "values", parameters will be passed as a list.
+    Otherwise, parameters will be expanded in the parameter list.
```

Comment:
I am a huge fan of having lots of examples. Just from reading those lines I get an idea of how it will behave but would have to read the code really carefully or test to fully understand. It would be awesome to have 2-3 examples that show exactly what will happen if you have `feature`/`parent`/`context` and/or `values`.

Reply:
That's actually a typo from a previous iteration. "values" no longer have any special treatment. (Replaced by `params_as_list`).

User typically would not use `register_function` directly, but rather the `@qgsfunction`decorator, so I'd rather keep the examples there.